lib/experimental/media/load.php

```diff
@@ -360,3 +360,28 @@ static function (): void {
 }
 
 add_action( 'wp_enqueue_media', 'gutenberg_override_media_templates' );
+
+/**
+ * Adds media-related settings to the block editor.
+ *
+ * @since X.X.X
+ *
+ * @param array $settings Existing block editor settings.
+ *
+ * @return array New block editor settings.
+ */
+function gutenberg_block_editor_settings_media_processing( $settings ) {
+	if ( ! gutenberg_is_experiment_enabled( 'gutenberg-media-processing' ) ) {
+		return $settings;
+	}
+
+	/** This filter is documented in wp-admin/includes/images.php */
+	$image_size_threshold = (int) apply_filters( 'big_image_size_threshold', 2560, array( 0, 0 ), '', 0 ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+
+	$settings['__experimentalAvailableImageSizes']   = gutenberg_get_all_image_sizes();
+	$settings['__experimentalBigImageSizeThreshold'] = $image_size_threshold;
+
+	return $settings;
+}
+
+add_filter( 'block_editor_settings_all', 'gutenberg_block_editor_settings_media_processing' );
```

Comment:
The Gutenberg team originally suggested moving these to the REST API instead of providing these as block editor settings.

See https://github.com/swissspidy/media-experiments/issues/561, https://github.com/swissspidy/media-experiments/pull/621 and https://github.com/swissspidy/media-experiments/blob/ec464e8a1a1e450a5f4610458a2d66ad3a141b31/inc/functions.php#L666-L721

The settings can then be retrieved via `getEntityRecord( 'root', '__unstableBase' )`

Reply:
I tried working on this a bit and ran into issues where importing the core datastore in the upload-media component seemed to cause a circular reference (PHP ran out of memory trying to load the page).

Any tips on where/how to add this or a link to the original suggestion?

Reply:
thanks, that confirms what I was thinking, I will give that a try.